lib/base/dependencygraph.hpp

```diff
@@ -21,12 +23,85 @@ class DependencyGraph
 	static void AddDependency(Object *parent, Object *child);
 	static void RemoveDependency(Object *parent, Object *child);
 	static std::vector<Object::Ptr> GetParents(const Object::Ptr& child);
+	static std::vector<Object::Ptr> GetChildren(const Object::Ptr& parent);
 
 private:
 	DependencyGraph();
 
+	/**
+	 * Represents an undirected dependency edge between two objects.
+	 *
+	 * It allows to traverse the graph in both directions, i.e. from parent to child and vice versa.
+	 */
+	struct Edge
+	{
+		// Note, due the existing confusing implementation of the DependencyGraph class, the terms parent and
+		// child are flipped here. Meaning, the parent is the dependent object and the child is the dependency.
+		// Example: A service object is dependent on a host object, thus the service object is the *parent* and
+		// the host object is the *child* one :).
+		Object* parent;
+		Object* child;
```

Comment:
💡 And, while on it, theoretically we could make them ConfigObject*.

Reply:
But only if that's not too complicated. 😅

Reply:
* Surprisingly, it isn't: #10264

Reply:
Surprisingly? :)